core/psh/ping/ping.c

```diff
@@ -103,36 +103,35 @@ static int psh_ping_socketConfig(void)
 }
 
 
-static void psh_ping_reqInit(uint8_t *data, int len)
+static void ping_reqinit(uint8_t *data, int len)
 {
-	struct icmphdr *hdr = (struct icmphdr *) data;
+	struct icmphdr *hdr = (struct icmphdr *)data;
```

Comment:
not related to this PR, but... this is possibly unaligned (`data` is `req` which is `uint8_t[64]` on stack with alignof(1), better to explicitly force alignment to 4

(this will probably solve other "unaligned access" patterns in this code, but there is symmetric problem with `resp`)

Reply:
Or even better, just declare `struct icmphdr` on stack and pass it to the function as `struct icmphdr *`, not an `uint8_t *`

Reply:
From what I see (https://github.com/phoenix-rtos/libphoenix/blob/6e57d09a03ed8c30af3751abf4c36fb9bc8af59a/include/netinet/ip_icmp.h#L25) `struct icmphdr` is of constant size, no elastic buffers there, so there should be no need to allocate it using byte array (or am I missing something?).

Reply:
we're appending additional data after the header (with variable size).

I would probably go for
```c
struct pkt {
  struct icmphdr;
  uint8_t data[0];
};
```

Current code allocates data by `malloc` so it's ok, but then passes data as `char*` - so we loose alignment information and sub-functions look like they might be invalid :).

No need to change it right away, but probably a good idea when we will be adding support for ipv6